src/util/sharing_node.h

```diff
@@ -151,8 +153,6 @@ SN_TYPE_PAR_DEF class sharing_node_innert : public sharing_node_baset
 
   d_it &write_internal()
   {
-    SN_ASSERT(data.use_count() > 0);
-
```

Comment:
Is there any test that shows that this is wrong (and is now correct)?

Reply:
This code was already exercised by several tests, but I used the wrong macro names to activate the sharing map/node internal assertions in the unit tests. Thus, none of the internal assertions were enabled and we didn't see the assertion failures. This is corrected by the first commit in this PR.